values.go

```diff
@@ -214,7 +215,11 @@ func (v String) MeteredType(common.MemoryGauge) Type {
 }
 
 func (v String) ToGoValue() any {
-	return string(v)
+	value := string(v)
+	if value == "" {
+		return nil
```

Comment:
Why return `nil` here for an empty string?

Reply:
I guess this is just a leftover from my blindly porting over the overflow code. The reasoning there was to make the output as terse as possible.

I can change this all the places here because I think that in cadence we should not make this distinction.